runtime/gc_modron_startup/mmparse.cpp

```diff
@@ -1247,6 +1247,25 @@ gcParseXXArguments(J9JavaVM *vm)
 		}
 	}
 
+	{
+		IDATA portableSharedCacheIndex = FIND_ARG_IN_VMARGS(EXACT_MATCH, VMOPT_XXPORTABLESHAREDCACHE, NULL);
+		IDATA noportableSharedCacheIndex = FIND_ARG_IN_VMARGS(EXACT_MATCH, VMOPT_XXNOPORTABLESHAREDCACHE, NULL);
+		if (portableSharedCacheIndex > noportableSharedCacheIndex) {
+			extensions->shouldForceToLowMemoryHeapCeilingShiftIfPossible = true;
+		}
+		else if (portableSharedCacheIndex < noportableSharedCacheIndex) {
+			extensions->shouldForceToLowMemoryHeapCeilingShiftIfPossible = false;
+		}
+		else {
```

Comment:
What this `else` for? How it is possible to have `portableSharedCacheIndex == noportableSharedCacheIndex` ?

Also please change formatting. Should be 
```
    if (....) {
        ....
    } else if () {
        ....
    } else if () {
        ....
    } else {
        ....
    }

Reply:
Also if we want this behaviour for containers I believe this is wrong place to set it. It should be set explicitly in `MM_Configuration::initializeRunTimeObjectAlignmentAndCRShift()`

Reply:
> Also if we want this behaviour for containers I believe this is wrong place to set it. It should be set explicitly in `MM_Configuration::initializeRunTimeObjectAlignmentAndCRShift()`. Parser should not make such decisions.
> I understand it is require to pass more information to recognize case `VMOPT_XXNOPORTABLESHAREDCACHE` is requested.

Okay I will move the decision to `initializeRunTimeObjectAlignmentAndCRShift()`.
```
		if (portableSharedCacheIndex != noportableSharedCacheIndex) {
			/* At least one option is set. Find the right most one. */
			if (portableSharedCacheIndex > noportableSharedCacheIndex) {
				extensions->portableSharedCacheEnabled = true;
			} else {
				extensions->portableSharedCacheEnabled = false;
			}
			extensions->portableSharedCacheExplicitlySet = true;
		} else {
			extensions->portableSharedCacheExplicitlySet = false;
		}
```
Though with that, one boolean variable won't be enough. The container decision is made whe user has not explicitly set the option.

Reply:
> Also this change means we always will run with shift 3 in any container run by default.

I think shift 3 should be used when the JVM is run in container AND uses AOT (for portability reasons)